core/src/cost_update_service.rs

```diff
@@ -174,6 +190,36 @@ impl CostUpdateService {
         );
         dirty
     }
+
+    // Based historical data, it seems reasonable to flag it as large change if updated_cost
+    // is more than double of input_data;
+    // Compare updated_cost against input_cost instead of previous calculated cost captures
+    // how far apart between single data point and calculated value, which is a good indicator
+    // of potential issues.
+    fn is_large_change_in_cost_calculation(input_cost: i64, updated_cost: i64) -> bool {
+        (updated_cost - input_cost).abs() > input_cost
```

Comment:
won't we get a bunch of these while a node is priming it's running average?  how tough would it be to plumb the tx max cost down here and only report when we get within like 90% of that?

Reply:
Yea, there will be more of this during the beginning of EMA, I looked at the `cndy` data, there would be 4,636 such cases out of 16,877 updates. 

Comparing calculation result against its input is bit fuzzy. I like your idea of only reporting when calculated results get close to MAX_UNITS, it also saved additional cost_model lock and lookup. I'll make the change

Reply:
actually I'll use MAX_UNITS after #23656, so the plumbing would be ready then

Reply:
Get back to this PR, here is my second thought: the purpose of this is 1) to log potential calculation anomaly, and when it does log, 2) it'd log exact data for the instance so we can back out the math. 

So, if it logs when calc result get close to compute_budget::MAX_UNITS, it doesn't really meet point 1. Some programs are legit expensive will be logged most of time, while other cheap program but had legit blip in calc will be skipped. Since the math is essentially `new_value = f(input_data, old_value)`, sounds like anomaly should be defined between input_data and new_value, therefore a better place to detect/report it should be within the math code itself (which also saves from additional map lookup mentioned above).

As for number of data sends to influx, regarding point 2, we don't want to aggregate it. To prevent from knocking out influx, ideally can implement moving window kind of thing -- can only submit X amount of data within Y seconds. But I feel that's bit of overkill, or implement it as part of `datapoint` function. 

wdyt?

Reply:
Sg. I think we can latch to new value N% delta on previous value and only the first occurrence of crossing the X% MAX_UNITS threshold.  Is that approximately what you're proposing?

Reply:
Close, but this still leave one case uncovered -- when calculation for a cheap program is unusual/suspicious. Say MAX_UNIT=1,400,000. If a program normally has 20,000, but a calc 10X it to 200,000, we probably want to capture this event, but since it is way below MAX_UNITS, it will not be logged. Feels like using MAX_UNITS not gonna work. 

Initially I expected simple changes for this PR. But to do it right, I'm down for this game plan: 
1. add calculation anomaly check to `execute_cost_table.upsert(...)`, it is where (most part of) math occurs, hence the right spot to report if there's data anomaly, regardless which algo (eg average, ema ...) is used. It can return an enum to indicate if large change of value is detected. 
2. the anomaly check could simply be ` new_calculated_value / input_data > 20` , it basically says it is unusual/interesting if aggregated result is 20X more than input data itself. I ran through data collected for `cndy`, that checks out. (for context, there were instances above math returns 4,400 when cndy went wrong)
3. to prevent writing too many data to influx too often, will implement a simple but generic version of throughput controller, that limits X influx insertions within 1 sec sliding window, more attempts beyond X will be dropped. 

(Turns out it's more fun than initially thought 😀 )

Reply:
Sorry I may not have been clear with my rehash, I was considering two bounds.
1) any time a given cost estimate is calculated to be E% above its previous value
2) the first time a given estimate crosses T% of MAX_UNITS (also gating `1`)

this would imply that after a metric crosses T% of MAX_UNITS, it will not publish metrics again.